### PR TITLE
Add missing genesis block number init

### PR DESCRIPTION
**Pull request by viquezclaudio**

### CODE REVIEW

File: blockchain/src/blockchain/blockchain.rs [hunk 1]
```diff
@@ -118,6 +118,12 @@ impl Blockchain {
             return Err(BlockchainError::InvalidGenesisBlock);
         }
 
+        let mut policy_config = Policy::default();
```
Comment (jsdanielh) [inline blockchain/src/blockchain/blockchain.rs]: We need to move this to the beginning

File: blockchain/src/blockchain/blockchain.rs [hunk 2]
```diff
@@ -118,6 +118,12 @@ impl Blockchain {
...
+        // The genesis block number must be set accordingly
+        policy_config.genesis_block_number = genesis_block.block_number();
+
+        let _ = Policy::get_or_init(policy_config);
```
Comment (jsdanielh) [inline blockchain/src/blockchain/blockchain.rs]: Could we error out if what we set is not what we got?

**Review by jsdanielh — CHANGES_REQUESTED** [review]
Also, there is a clippy warning

**Review by jsdanielh — CHANGES_REQUESTED** [review]
We need to do this also in `spammer`

**Comment by viquezclaudio** [comment]
Its not needed for the spammer because the spammer is calling `Client = Client::from_config`

**Comment by jsdanielh** [comment]
> Its not needed for the spammer because the spammer is calling `Client = Client::from_config`

Yeah, you're right. But I think I found a place where it is needed: `zkp/examples/prover/prove.rs`

**Review by jsdanielh — APPROVED** [review]

**Review by jsdanielh — APPROVED** [review]

**Pull request merged**